irony-cdb.el

```diff
@@ -114,6 +128,24 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--locate-dominating-file-with-dirs (file name subdirectories)
+  "Convenience wrapper around `locate-dominating-file'
+
+Looks up the directory hierarchy from FILE for to locate any directory
+in `subdirectories` which contains NAME. Stops at the first parent
+directory where the test succeeds and returns the full path to file
+found."
+  (let ((check-subdirs
```

Comment:
Good catch, I did not know about version availability. hmm.

Ok. We iterate over `irony-cdb-search-directory-list`, call `locate-dominating-file` for each 
one and end up with a list of 0 or more candidates.

How would you like to choose which one? in most cases, there'll only be one. If there are more,
the one with the most delimiters is a good heuristic for "most specific", or we could just take the first one and punt. Which do you prefer?